lib/Doctrine/ORM/Mapping/ManyToManyOwningSideMapping.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+use function assert;
+use function strtolower;
+use function trim;
+
+final class ManyToManyOwningSideMapping extends ManyToManyAssociationMapping implements AssociationOwningSideMapping
+{
+    /**
+     * Specification of the join table and its join columns (foreign keys).
+     * Only valid for many-to-many mappings. Note that one-to-many associations
+     * can be mapped through a join table by simply mapping the association as
+     * many-to-many with a unique constraint on the join table.
+     */
+    public JoinTableMapping $joinTable;
+
+    /** @return mixed[] */
+    public function toArray(): array
+    {
+        $array = parent::toArray();
+
+        $array['joinTable'] = $this->joinTable->toArray();
+
+        return $array;
+    }
+
+    /** @param mixed[] $mappingArray */
+    public static function fromMappingArrayAndNamingStrategy(array $mappingArray, NamingStrategy $namingStrategy): self
+    {
+        $mapping = parent::fromMappingArray($mappingArray);
+
+        // owning side MUST have a join table
+        if (! isset($mapping['joinTable']['name'])) {
+            if (! isset($mapping->joinTable)) {
+                $mapping->joinTable = new JoinTableMapping();
+            }
+
+            $mapping->joinTable['name'] = $namingStrategy->joinTableName(
```

Comment:
I noticed that you're switching between array access and properties within the new DTOs. I understand that keeping array access keeps the diff small for old code, but inside the new classes, I believe using properties consistently would improve the readability. Especially in methods where we map from arrays to DTOs and vice versa, using properties would make it clearer what is the array and what the DTO.

In this very line, you're even using both methods in one expression. 😵